File: chapter06/aakkademaid-java-6/src/main/java/com/lp/akka/notes/ParseArticleApp.java
package com.lp.akka.notes;

import akka.actor.ActorRef;
import akka.actor.ActorSystem;
import akka.actor.Props;
import akka.contrib.pattern.ClusterReceptionistExtension;
import akka.routing.BalancingPool;

/**
 *  使用Router Pool 中用于解析文章的 Actor 放到集群上，然后启动集群。
 *  启动完成后，我们就将介绍如何从另一个 Actor 系统访问集群中的服务
 */
public class ParseArticleApp {
    public static void main(String... args) {
        ActorSystem system = ActorSystem.create("lp-akka-cluster");

        // 使用BalancingPool构建多个routee   // router at /user/workers
        ActorRef workers =
                system.actorOf(
                        new BalancingPool(5).props(Props.create(ArticleParseActor.class)),
                        "workers"
                );

        // 在 ClusterReceptionist 中注册 worker Actor
        ((ClusterReceptionistExtension) akka.contrib.pattern.ClusterReceptionistExtension.apply(system)).
                registerService(workers);
    }
}

File: chapter06/aakkademaid-scala-6/src/main/scala/com/lp/akka/notes/ParseArticleApp.scala
package com.lp.akka.notes

import akka.actor.{ActorSystem, Props}
import akka.contrib.pattern.ClusterReceptionistExtension
import akka.routing.BalancingPool

/**
 * @title TODO
 * @author li.pan
 * @version 1.0.0
 * @createTime 2021年04月30日 14:42:00
 *             <p>
 *             使用Router Pool 中用于解析文章的 Actor 放到集群上，然后启动集群。
 *             启动完成后，我们就将介绍如何从另一个 Actor 系统访问集群中的服务
 *             </p>
 */
object ParseArticleApp extends App {

  val system = ActorSystem("Akkademy")
  val clusterController = system.actorOf(Props[ClusterController], "clusterController")

  // router at /user/workers
  val workers =
    system.actorOf(BalancingPool(5).props(Props[ArticleParseActor]), "workers")

  // 在 ClusterReceptionist 中注册 worker Actor
  ClusterReceptionistExtension(system).registerService(workers)
}
